functions/wav_time.h

```diff
@@ -0,0 +1,143 @@
+#ifndef FUNCTIONS_WAV_TIME_H
+#define FUNCTIONS_WAV_TIME_H
+
+#include "wav_effect.h"
+#include "int.h"
+#include "../blades/blade_base.h"
+
+// Usage: WavLen<EFFECT_NAME>  
+// EFFECT_NAMENUM : any off the available effect form the style edittor
+// Returns length of wav file returned as const int in [ms]
+// returned value: INTEGER
+//
+template<BladeEffectType EFFECT>
+class WavLen {
+ public:
+  void run(BladeBase* blade) {
+    found_ = false;
+    eff_name_ = wav_effect.GetEffectName(EFFECT);//find correspondig SoundEffectName for BladeEffectType
+    for (size_t i = 0; i < NELEM(wav_players); i++) {
+        if(strstr(eff_name_, wav_players[i].effectname()) && wav_players[i].isPlaying()) {found_ = true; idx=(int)i;}
```

Comment:
Interacting with the wav players can be dangerous, because they are updated by interrupts and so these functions can change their values in the middle of executing this code, which makes it really hard to know if this will work as intended or not. Especially if the compiler decides to reorder things.

Reply:
Every loop, the wavplayers are checked to find were the current sound is playing. even if the order or wavplayer number would change while playing the sound, we still would find the correct one right?